fish-rust/src/parse_constants.rs

```diff
@@ -90,7 +90,6 @@ mod parse_constants_ffi {
 
     /// IMPORTANT: If the following enum table is modified you must also update token_type_description below.
     /// TODO above comment can be removed when we drop the FFI and get real enums.
-    #[derive(Clone, Copy)]
```

Comment:
I think leaving these here for clarity, even if unnecessary, doesn't hurt.

Reply:
`Clone` and `Copy` are useful to define because you never know when you might need them later on and they make it clear that the developer considers a clone to be a low-cost operation (without having to review the source). 

`Debug` and `Default` are iffier since (in large quantities) they can slow down the compiler if you don't need them, but it's a real chore to go back and add them everywhere as well, so I'm still undecided on whether or not it makes sense to sprinkle them everywhere liberally.

Reply:
> I think leaving these here for clarity, even if unnecessary, doesn't hurt.

then maybe add all the `Copy, Clone, Eq, and PartialEq` since they are there?